app/dashboard/router.py

```diff
@@ -269,6 +270,18 @@ def get_queryset(self):
             end = start + int(limit)
             queryset = queryset[start:end]
 
+        # workaround for https://github.com/gitcoinco/web/issues/3348
+        pks = queryset.values_list('pk', flat=True)
+        queryset = Bounty.objects.filter(pk__in=pks)
```

Comment:
I don't think we should need to perform a whole new query here 🤔  Looks like the shelf life of our logic has been exhausted. If we weren't slicing the QS on [line 271](https://github.com/gitcoinco/web/pull/3364/files#diff-a0ceb478149ae8595c669666edec2decR271), this wouldn't be an issue.  Thoughts on eliminating this in favor of standard DRF pagination/sorting/filtering?

Reply:
whats the time estimate on that?  i haven't seen much action on the refactor ticket, don't even know if it's current

Reply:
I am not sure of the time estimate. If it is a priority I can also work on the DRF. Even if not now, DRF pagination does sound the way to go. Even if not that, how about storing the non-sliced QS in another variable `queryset_full` and then later just do `queryset = queryset_full`. At least that will reduce the number of queries to the DB which is usually a very costly operation. cc @mbeacom @owocki